cmake/abseil-cpp.cmake

```diff
@@ -39,7 +39,7 @@ set(_protobuf_FIND_ABSL "if(NOT TARGET absl::strings)\n  find_package(absl CONFI
 
 if (BUILD_SHARED_LIBS AND MSVC)
   # On MSVC Abseil is bundled into a single DLL.
-  set(protobuf_ABSL_USED_TARGETS abseil_dll)
+  set(protobuf_ABSL_USED_TARGETS absl::abseil_dll)
```

Comment:
it looks like this breaks out cmake+msvc shared library build.  What version of cmake are you using where this is necessary?

Reply:
3.24

Reply:
Oh I see the issue, this change is only required when using an abseil that is not part of your git submodule system. I'll add the necessary guards.

Reply:
ah that makes a lot of sense, thanks for tracking that down!  Long-term we'd like to remove these submodules, and definitely not use them for testing.  Things like this keep popping up :(

Reply:
I'll look into that as it has implications for my use case as well! I'd expect abseil just doesn't export those namespaces for MSVC because it only creates that single target or something like that.

Submodules are a bear to maintain to be sure, particularly on top of multiple possible methods of providing the deb the submodule vendors! 
Out of curiosity, what's the blocker for making the switch?

Reply:
There's no real blocker, just prioritization.  We're working with folks from grpc and abseil to try to unify our cmake strategy.  The end result will likely end up with use deleting these submodules, or just keeping them around for developers (and maybe a single dedicated CI test for them)

Reply:
Ah, very cool! The submodules are kinda nice to "just have" the dependency, but it's a lot more work to be sure.
FWIW I truly appreciate the ability to easily bring in an external abseil/etc with just the `find_package` command, makes everything much simpler, particularly coming from a package manager env like [spack](https://github.com/spack/spack) (which is the driver behind this PR). The find_package interface is one of CMake's most useful features IMO.

Reply:
yea we're likely going to lean more heavily on `find_package` (and test it more thoroughly)

Reply:
FYI, it's best CMake practice (according to Craig Scott's professional CMake book) to have the target names the same for either a `find_package` or `add_subdirectory` method. 

In the install, the export's namespace `ns` should match any call to `add_librarary(ns::the_targetALIAS the_target)`. Then, consumers will always link to `ns::the_target` regardless of how it the target was found.